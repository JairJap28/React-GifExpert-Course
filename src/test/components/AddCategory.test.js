import React from 'react';
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Test on AddCategory Component', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Should be shown correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should change the input box', () => {
        const input = wrapper.find('input');
        const value = 'Hello World';

        input.simulate('change', { target: { value } });
    });

    test('Should not post info on Submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Should post info on Submit', () => {
        wrapper.find('input').simulate('change', { target: { value: 'Testing components' } });
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});