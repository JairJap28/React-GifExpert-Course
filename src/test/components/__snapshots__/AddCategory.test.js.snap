// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on AddCategory Component Should be shown correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
